tests/components/jewish_calendar/test_binary_sensor.py

```diff
@@ -0,0 +1,107 @@
+
+    melacha_params = [
+        make_nyc_test_params(dt(2018, 9, 1, 16, 0), STATE_ON),
+        make_nyc_test_params(dt(2018, 9, 1, 20, 21), STATE_OFF),
+        make_nyc_test_params(dt(2018, 9, 7, 13, 1), STATE_OFF),
+        make_nyc_test_params(dt(2018, 9, 8, 21, 25), STATE_OFF),
+        make_nyc_test_params(dt(2018, 9, 9, 21, 25), STATE_ON),
+        make_nyc_test_params(dt(2018, 9, 10, 21, 25), STATE_ON),
+        make_nyc_test_params(dt(2018, 9, 28, 21, 25), STATE_ON),
+        make_nyc_test_params(dt(2018, 9, 29, 21, 25), STATE_OFF),
+        make_nyc_test_params(dt(2018, 9, 30, 21, 25), STATE_ON),
+        make_nyc_test_params(dt(2018, 10, 1, 21, 25), STATE_ON),
+        make_jerusalem_test_params(dt(2018, 9, 29, 21, 25), STATE_OFF),
+        make_jerusalem_test_params(dt(2018, 9, 30, 21, 25), STATE_ON),
+        make_jerusalem_test_params(dt(2018, 10, 1, 21, 25), STATE_OFF),
+    ]
+    melacha_test_ids = [
+        "currently_first_shabbat",
+        "after_first_shabbat",
+        "friday_upcoming_shabbat",
+        "upcoming_rosh_hashana",
+        "currently_rosh_hashana",
+        "second_day_rosh_hashana",
+        "currently_shabbat_chol_hamoed",
+        "upcoming_two_day_yomtov_in_diaspora",
+        "currently_first_day_of_two_day_yomtov_in_diaspora",
+        "currently_second_day_of_two_day_yomtov_in_diaspora",
+        "upcoming_one_day_yom_tov_in_israel",
+        "currently_one_day_yom_tov_in_israel",
+        "after_one_day_yom_tov_in_israel",
+    ]
+
+    @pytest.mark.parametrize(
+        [
+            "now",
+            "candle_lighting",
+            "havdalah",
+            "diaspora",
+            "tzname",
+            "latitude",
+            "longitude",
+            "result",
+        ],
+        melacha_params,
+        ids=melacha_test_ids,
+    )
+    async def test_issur_melacha_sensor(
+        self,
+        hass,
+        now,
+        candle_lighting,
+        havdalah,
+        diaspora,
+        tzname,
+        latitude,
+        longitude,
+        result,
+    ):
+        """Test Issur Melacha sensor output."""
+        time_zone = get_time_zone(tzname)
+        test_time = time_zone.localize(now)
+
+        default_time_zone = str(hass.config.time_zone)
+        hass.config.set_time_zone(tzname)
+        hass.config.latitude = latitude
+        hass.config.longitude = longitude
+
+        assert await async_setup_component(
+            hass,
+            jewish_calendar.DOMAIN,
+            {
+                "jewish_calendar": {
+                    "name": "test",
+                    "language": "english",
+                    "diaspora": diaspora,
+                    "candle_lighting_minutes_before_sunset": candle_lighting,
+                    "havdalah_minutes_after_sunset": havdalah,
+                }
+            },
+        )
+        await hass.async_block_till_done()
+
+        with alter_time(test_time):
+            await hass.helpers.entity_component.async_update_entity(
```

Comment:
Fire a time change event instead to move time forward while patching time.

https://github.com/home-assistant/home-assistant/blob/3c07a9b4c75bb3baaf132018ba322a8bc79b8103/tests/common.py#L277

Reply:
I could do that, but I still would need to patch time so I get the correct results for the test as the results are time-sensitive.

Would you prefer:
```
 with alter_time(test_time):
       future = dt_util.utcnow() + timedelta(seconds=30)
       async_fire_time_changed(hass, future)
       await hass.async_block_till_done()
```

just to get the async_update to trigger?

Reply:
Probably use `test_time` instead of utcnow since we're patching utcnow.

Reply:
I think I can simplify the code as I don't think I need to patch utcnow.

Reply:
We should always patch time when we want to control time.